packages/portfolio-planner/src/engine.ts

```diff
@@ -2,7 +2,12 @@
 /* eslint-env node */
 import { inspect } from 'node:util';
 import { q, Fail } from '@endo/errors';
-import { fromUniqueEntries } from '@agoric/internal';
+import { mustMatch } from '@endo/patterns';
```

Comment:
consider using the one that works with `TypedPattern<T>`:

```suggestion
import { mustMatch } from '@agoric/internal';
```